web/src/pm_list_data.ts

```diff
@@ -60,7 +61,13 @@ export function get_conversations(): DisplayObject[] {
         conversations.unshift({user_ids_string: active_user_ids_string, max_message_id: -1});
     }
 
-    for (const conversation of conversations) {
+    const filtered_conversations = util.filter_by_word_prefix_match(
+        conversations,
+        search_term,
+        (item) => people.get_recipients(item.user_ids_string),
+    );
```

Comment:
How did you select this filtering function? `typeahead_helper.query_matches_person` and `filter_people_by_search_terms` are the two existing functions we use to match users in typeahead and the right sidebar buddy list, and I don't think it makes sense to add a third variant that is less specialized to the case of matching users.

I think my leaning would be to use `filter_people_by_search_terms`, since that's designed for this specific filtering context, with helpful features like ignoring diacritics in the matching.

Reply:
> How did you select this filtering function?

I was replicating the functionality of topic_list_data where this function was being used, but since the data is quite different from a single string topic, in hindsight I should have looked for other options. 

I've changed the function!

I've also made some changes to the tests as described in https://github.com/zulip/zulip/pull/30332#discussion_r1648380265